From 7c2ca1e348437925e0c1d9a260da269caf6160ff Mon Sep 17 00:00:00 2001
From: Michael Scott <mike@foundries.io>
Date: Sat, 6 Nov 2021 16:05:37 -0700
Subject: [PATCH 22/45] portenta-x8: cleanup header config

Changes of note:
- LOADADDR reverted to original value
- SYS_MALLOC_LEN to 128MB (this fixes crash in Fedora install)

Signed-off-by: Michael Scott <mike@foundries.io>
---
 include/configs/portenta-x8.h | 12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)

diff --git a/include/configs/portenta-x8.h b/include/configs/portenta-x8.h
index 7ff4f8a30c..bc785e0356 100644
--- a/include/configs/portenta-x8.h
+++ b/include/configs/portenta-x8.h
@@ -50,7 +50,6 @@
 #undef CONFIG_SPL_DM_I2C
 
 #define CONFIG_SYS_I2C
-#define CONFIG_SYS_I2C_SPEED      100000
 
 #if defined(CONFIG_NAND_BOOT)
 #define CONFIG_SPL_NAND_SUPPORT
@@ -156,8 +155,6 @@
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE "\0" \
 	"fdt_file=" CONFIG_DEFAULT_FDT_FILE "\0" \
 	"bootm_size=0x10000000\0" \
-	"initrd_addr=0x43800000\0"		\
-	"initrd_high=0xffffffffffffffff\0" \
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
 	"mmcpart=" __stringify(CONFIG_SYS_MMC_IMG_LOAD_PART) "\0" \
 	"mmcroot=" CONFIG_MMCROOT " rootwait rw\0" \
@@ -213,7 +210,7 @@
 #endif
 
 /* Link Definitions */
-#define CONFIG_LOADADDR			0x43800000
+#define CONFIG_LOADADDR			0x40800000
 
 #define CONFIG_SYS_LOAD_ADDR		CONFIG_LOADADDR
 
@@ -234,7 +231,7 @@
 #define CONFIG_MMCROOT			"/dev/mmcblk2p2"  /* USDHC3 */
 
 /* Size of malloc() pool */
-#define CONFIG_SYS_MALLOC_LEN		SZ_32M
+#define CONFIG_SYS_MALLOC_LEN		SZ_128M
 
 #define CONFIG_SYS_SDRAM_BASE           0x40000000
 #define PHYS_SDRAM                      0x40000000
@@ -255,11 +252,7 @@
 /* USDHC */
 #define CONFIG_FSL_USDHC
 
-#ifdef CONFIG_TARGET_IMX8MM_DDR4_EVK
-#define CONFIG_SYS_FSL_USDHC_NUM	1
-#else
 #define CONFIG_SYS_FSL_USDHC_NUM	2
-#endif
 #define CONFIG_SYS_FSL_ESDHC_ADDR       0
 
 #define CONFIG_SYS_MMC_IMG_LOAD_PART	1
@@ -297,7 +290,6 @@
 #define CONFIG_VIDEO_LOGO
 #define CONFIG_SPLASH_SCREEN
 #define CONFIG_SPLASH_SCREEN_ALIGN
-#define CONFIG_CMD_BMP
 #define CONFIG_BMP_16BPP
 #define CONFIG_BMP_24BPP
 #define CONFIG_BMP_32BPP
-- 
2.35.1

